### ISSUE TITLE
[$250] Task - As assignee if user taps Mark as complete, displays error message

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38472]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.53 
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Launch app
2. Tap fab -- Assign task
3. Enter title
4. Select a user for somewhere
5. Select another user as Assignee
6. Tap confirm task
7. Go to https://staging.new.expensify.com/
in mweb
8. Login as task assignee
9. Tap on the newly assigned task report from LHN
10. Tap Mark as complete

## Expected Result:
As assignee if user taps Mark as complete, must not display any error message .

## Actual Result:
As assignee if user taps Mark as complete, displays error message " You do not have permission to do the requested action"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/4a3f03bc-a911-449a-9306-7e7739397d76

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01979d99f098439a46</li>
        <li>Upwork Job ID: 1778528383583100928</li>
        <li>Last Price Increase: 2024-08-08</li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
We think this issue might be related to the #vip-vsb.

**Comment by isabelastisser** [comment]
Reviewing it.

**Comment by isabelastisser** [comment]
I'm having issues adding this issue here:

https://github.com/orgs/Expensify/projects/106/views/1

try again tonight.

**Comment by isabelastisser** [comment]
Added to VIP-VSB.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by abdulrahuman5196** [comment]
no proposals to review

**Comment by isabelastisser** [comment]
Still waiting for proposals.

**Comment by isabelastisser** [comment]
This has been added to the #vip-vsb project as a medium priority.

**Comment by isabelastisser** [comment]
No proposals yet.

**Comment by abdulrahuman5196** [comment]
No proposals yet

**Comment by isabelastisser** [comment]
It is not overdue; we are waiting for proposals.

@abdulrahuman5196, can we still reproduce this? If not, let's close it.

**Comment by isabelastisser** [comment]
I will try to reproduce this later today.

**Comment by isabelastisser** [comment]
I will prioritize this soon.

**Comment by rezkiy37** [comment]
Hi, I’m Michael (Mykhailo) from Callstack and I would like to work on this issue.

**Comment by rezkiy37** [comment]
I confirm that the bug is reproducible. I've investigated the issue. It seems like a backend problem as well. The bug happens if an assigned user is not a member of a report where a task was shared. So a sender should invite the assigned user first and only after that assign a task. There 2 ways to fix it:
1. Allow the assigned user to complete tasks even when it is in a foreign report. 
2. Restrict any actions with a task if the assigned user is not a report member.

I prefer the 1st variant to the 2nd one. We need a backend engineer here. Thank you!

<details><summary>Screenshots</summary>
<p>

<img width="1723" alt="Screenshot 2024-05-06 at 16 33 51" src="https://github.com/Expensify/App/assets/57314631/63cf0e22-3b00-4fb8-94b5-3fef01c4c816">
<img width="1114" alt="Screenshot 2024-05-06 at 16 35 57" src="https://github.com/Expensify/App/assets/57314631/733d9b81-f90d-466f-b43c-c478b1060df3">


</p>
</details>

**Comment by isabelastisser** [comment]
Not overdue, waiting for internal engineering assignment.

**Comment by isabelastisser** [comment]
Needs internal engineering assignment.

**Comment by isabelastisser** [comment]
Still waiting for proposals.

**Comment by rezkiy37** [comment]
> Still waiting for proposals.

Do you mean an internal engineer?

**Comment by rezkiy37** [comment]
Not overdue, waiting for internal engineering assignment.

**Comment by isabelastisser** [comment]
Not overdue, waiting for internal engineering assignment.

**Comment by isabelastisser** [comment]
Posted to #vip-vsb again now.

https://expensify.slack.com/archives/C066HJM2CAZ/p1716221484778999

**Comment by isabelastisser** [comment]
De-prioritizing this based on this discussion.

https://expensify.slack.com/archives/C066HJM2CAZ/p1716510847407189

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by rezkiy37** [comment]
Hello!
FYI: I have a vacation from 11.07 to 26.07. Meanwhile, someone from Callstack can help with this issue.

**Comment by rezkiy37** [comment]
@isabelastisser hey! Any updates? Do we have a developer for this issue?

**Comment by isabelastisser** [comment]
Hey, @rezkiy37! This is not a priority, so no updates.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by rezkiy37** [comment]
I can confirm the bug is not replicable anymore. We can close the issue. 

https://github.com/user-attachments/assets/a5e3534e-21da-47f4-ba0a-873031ae56af

cc @isabelastisser @abdulrahuman5196 @mvtglobally @izarutskaya

**Comment by isabelastisser** [comment]
Thanks, @rezkiy37! Closing.
